tests/jballerina-unit-test/src/test/resources/test-src/object/ObjectProject/src/init/object-initializer.bal

```diff
@@ -398,3 +398,41 @@ function testInitInvocationWithCheckAndRestParams2() returns (boolean) {
     var [s, marksBeforeChange, marksAfterChange] = testInitInvocationWithCheckAndRestParams(10, ...modules);
     return !(s is error) && marksBeforeChange == 90 && marksAfterChange == 95;
 }
+
+type Student6 object {
```

Comment:
Do we already have test cases for non-trivial expressions as default values? If not, shall we add them as well? For example, object init, mapping constructor, anonymous functions, type test expressions etc.